lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -29,6 +32,144 @@
  * of patches to that input. Tokenizes HTML but does not fully
  * parse the input document.
  *
+ * ## Usage
+ *
+ * Use of this class requires three steps:
+ *
+ *  1. Create a new class instance with your input HTML document.
+ *  2. Find the tag(s) you are looking for.
+ *  3. Request changes to the attributes in those tag(s).
+ *
+ * Example
+ * ```php
+ *     $processor = new WP_HTML_Tag_Processor( $html );
+ *     if ( $processor->next_tag( array( 'tag_name' => 'option' ) ) ) {
+ *         $processor->set_attribute( 'selected', true );
+ *     }
+ * ```
+ *
+ * ### Finding tags
+ *
+ * The `next_tag()` function moves the internal cursor through
+ * your input HTML document until it finds a tag meeting any of
+ * the supplied restrictions in the optional query argument. If
+ * no argument is provided then it will find the next HTML tag,
+ * regardless of what kind it is.
+ *
+ * If you want to _find whatever the next tag is_
+ * ```php
+ *     $processor->next_tag();
+ * ```
+ *
+ * | Goal                                                      | Query                                                                              |
+ * |-----------------------------------------------------------|------------------------------------------------------------------------------------|
+ * | Find any tag.                                             | `$processor->next_tag();`                                                          |
+ * | Find next image tag.                                      | `$processor->next_tag( array( 'tag_name' => 'img' ) );`                            |
```

Comment:
```suggestion
  * | Find next image tag.                                      | `$processor->next_tag( 'img' );`                                                   |
  * | Find next image tag.                                      | `$processor->next_tag( array( 'tag_name' => 'img' ) );`                            |
```

Reply:
see my previous comment on this. I have second-guesses about this short-hand

Reply:
Wait, do you mean second thoughts on surfacing it here, or second thoughts on even shipping it in this code?

Reply:
supporting it at all. my worry is that we say you can search for a given tag name, but then people want a short hand for class name, so we add it, but then if you don't specify a tag you have `->next( null, 'wp-block-group' )` everywhere. or someone things, let's put class name first, and then we have the other problem.

I'd even be for exposing this shorthand only after we see use in practice to see if people even want it, and if so, which convention is more dominant.

alternatively it's always easy to later add new specialized methods. suppose we only expose `->next()` right now, taking a query object. we can add `->next_tag( 'img' )` and `->next_with_class( 'wp-block-group' )`. performance wise this is a "free" option

Reply:
> I'd even be for exposing this shorthand only after we see use in practice to see if people even want it, and if so, which convention is more dominant.

Let's do that 👍